### PR TITLE
[PDI-18692] Add site files to embedded metastore of KTR/KJB files.

### PR DESCRIPTION
**Pull request by tkafalas**

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
:ok_hand: All tests passed!
 
Tests run: **346**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-2039/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
:ok_hand: All tests passed!
 
Tests run: **346**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-2039/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
:ok_hand: All tests passed!
 
Tests run: **346**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-2039/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
:ok_hand: All tests passed!
 
Tests run: **346**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-2039/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterImpl.java [hunk 1]
```diff
@@ -598,37 +608,85 @@ public String toXmlForEmbed( String rootTag ) {
...
-  private Node createChildElement( Document doc, String elementName, String elementType, String elementValue ) {
-    Element childNode = doc.createElement( "child" );
-    childNode.appendChild( createTextNode( doc, "id", elementName ) );
-    childNode.appendChild( createTextNode( doc, "value", elementValue ) );
+  private Object unmarshallSiteFileNode( Node field ) {
+    ArrayList<NamedClusterSiteFile> namedClusterSiteFiles = new ArrayList<>();
+    Node siteFileWrapper = XMLHandler.getSubNode( field, CHILDREN );
+    if ( siteFileWrapper != null ) {
+      List<Node> siteFilesNode = XMLHandler.getNodes( siteFileWrapper, CHILD );
+      for ( Node siteFile : siteFilesNode ) {
```
Comment (mkambol) [inline impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterImpl.java]: Can you extract some of this to not be quite so deeply nested?  Once I hit level four of a nesting my eyes start to glaze over.

**Comment by buildguy** [comment]
### :x: Build failed in 35s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl impl/cluster
```
---
:ok_hand: All tests passed!
 
Tests run: **346**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-plugin/job/PR-2039/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**